src/state-card-custom-ui.html

```diff
@@ -30,36 +30,76 @@
     'inputChanged(hass, inDialog, stateObj)',
   ],
 
-  badgeMode: function (hass, stateObj) {
+  badgeMode: function (hass, stateObj, badgeLabelBorder, badgeLabelText, badgeBackground) {
     var states = [];
-    stateObj.attributes.entity_id.forEach((id) => {
-      var state = hass.states[id];
-      if (!state) {
-        /* eslint-disable no-console */
-        console.warn('Unknown ID ' + id + ' in group ' + stateObj.entity_id);
-        /* eslint-enable no-console */
-        return;
-      }
-      if (!stateObj.attributes.badges_list ||
-          stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
-        states.push(window.customUI.maybeChangeObject(
-          this, state, false /* inDialog */, false /* allowHidden */));
-      }
-    });
+    if ( typeof stateObj.attributes.entity_id !== 'undefined' && window.hassUtil.computeDomain(stateObj) === 'group' ) {
+      stateObj.attributes.entity_id.forEach((id) => {
+        var state = hass.states[id];
+        if (!state) {
+          /* eslint-disable no-console */
+          console.warn('Unknown ID ' + id + ' in group ' + stateObj.entity_id);
+          /* eslint-enable no-console */
+          return;
+        }
+        if (!stateObj.attributes.badges_list ||
+            stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
+          states.push(window.customUI.maybeChangeObject(
+            this, state, false /* inDialog */, false /* allowHidden */));
+        }
+      });
+    } else {
+        var state = hass.states[stateObj.entity_id];
+        if (!state) {
+          /* eslint-disable no-console */
+          console.warn('Unknown ID ' + stateObj.entity_id + ' in group ' + stateObj.entity_id);
+          /* eslint-enable no-console */
+          return;
+        }
+        if (!stateObj.attributes.badges_list ||
+            stateObj.attributes.badges_list.indexOf(state.entity_id) >= 0) {
+          states.push(window.customUI.maybeChangeObject(
+            this, state, false /* inDialog */, false /* allowHidden */));
+        }
+    }
     window.hassUtil.dynamicContentUpdater(
       this,
       'HA-BADGES-CARD',
       { hass: hass, states: states });
     Polymer.dom(this).lastChild.style.setProperty('--ha-state-label-badge-margin-bottom', '0');
+    Polymer.dom(this).lastChild.style.setProperty('--label-badge-red', badgeLabelBorder);
+    Polymer.dom(this).lastChild.style.setProperty('--label-badge-text-color', badgeLabelText);
```

Comment:
In the default case this will result in `style="--label-badge-text-color: var(--label-badge-text-color)"`.
Can this be avoided?

Reply:
I suppose we can prevent this from happening by adding some checks;

    if (badgeLabelBorder !== 'var(--label-badge-red)') {
      Polymer.dom(this).lastChild.style.setProperty('--label-badge-red', badgeLabelBorder);
    }
    if (badgeLabelText !== 'var(--label-badge-text-color)') {
      Polymer.dom(this).lastChild.style.setProperty('--label-badge-text-color', badgeLabelText);
    }
    if (badgeBackground !== 'var(--label-badge-background-color)') {
      Polymer.dom(this).lastChild.style.setProperty('--label-badge-background-color', badgeBackground);
    }}

Reply:
Ok, as you suggest in your later comment, I think it's better to leave this as it was.
The check described above only functions in the very unlikely case that a user has added the custom_ui and `var(--label-badge-red)` attributes and even then it won't hurt... These checks just adds verbosity to the code without much use then.